### PR TITLE
Jan2

### PR DESCRIPTION
**Pull request by muhammadanas123**

### CODE REVIEW

File: app/assets/stylesheets/custom.css.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    
+    // }
+    
+     
+    
+    // h1 {
+    
+    // color: red;
+    
+    // }
```
Comment (UmerQaisar) [inline app/assets/stylesheets/custom.css.scss:27]: Please remove all commented code from file. @muhammadanas123

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class HomeController < ApplicationController
+    def index
```
Comment (UmerQaisar) [inline app/controllers/home_controller.rb:2]: remove extra spaces and indent properly. @muhammadanas123

File: app/assets/stylesheets/custom.css.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@import 'bootstrap/dist/css/bootstrap';
+
+// .navbar {
```
Comment (momnanaseer) [inline app/assets/stylesheets/custom.css.scss:3]: Remove all commented code from file @muhammadanas123

File: app/controllers/home_controller.rb (hunk 2 above)
Comment (momnanaseer) [inline app/controllers/home_controller.rb:2]: remove extra spaces and indent properly  @muhammadanas123 
def index; end

File: app/controllers/journeys_controller.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    def show
+
+
+    end
+
+
+    def index
+        # byebug
+        @traveller = Traveller.find(params[:traveller_id])
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:31]: use find_by instead of find. @muhammadanas123

File: app/controllers/journeys_controller.rb [hunk 5]
```diff
@@ -0,0 +1,101 @@
+class JourneysController < ApplicationController
+
+    before_action :traveller_cannot_access_the_other_traveller_info, only: [:edit, :show, :update, :destroy]
+    before_action :find_and_set_journey, only: [:show, :edit, :update, :destroy]
+    before_action :restrict_sender_to_access_journey
+
+
+    def new
+        # byebug
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:9]: remove commeted byebug.

File: app/controllers/journeys_controller.rb [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    def create 
+        @traveller = Traveller.find(params[:traveller_id])
+        @journey = @traveller.journeys.create(journey_params)
+        redirect_to traveller_journeys_path(@traveller, @journey), notice: "successfully created a journey"
+        # render json: @journey
+
+    end
+
+    def show
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:23]: remove extra spaces, if action is empty then we can also write it like this
def show; end

File: app/controllers/journeys_controller.rb [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    end
+
+
+
+
+
+    private
+
+    def find_and_set_journey
+        @traveller = Traveller.find(params[:traveller_id])
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:76]: use find_by instead of find

File: app/controllers/journeys_controller.rb [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+    def edit
+
+    end
+
+    def update
+        # byebug
+        from = @journey.from
+        to = @journey.to
+
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:43]: I dont think we need conditions, `@journey.update(journey_params)` should serve the purpose.

File: app/controllers/journeys_controller.rb [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        elsif journey_params[:from] == ""
+            @journey.update(from: from, to: journey_params[:to], departure_date: journey_params[:departure_date], arrival_date: journey_params[:arrival_date], capacity: journey_params[:capacity], rate: journey_params[:rate])
+        elsif journey_params[:to] == ""
+            @journey.update(from: journey_params[:from], to: to, departure_date: journey_params[:departure_date], arrival_date: journey_params[:arrival_date], capacity: journey_params[:capacity], rate: journey_params[:rate])
+        else
+            @journey.update(journey_params)
+        end
+
+
+        if @journey.errors.any?
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:55]: also dont need this condition
```
if @journey.update(journey_params)
   flash message 
else
     render :edit
end
```

File: app/controllers/journeys_controller.rb [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    def find_and_set_journey
+        @traveller = Traveller.find(params[:traveller_id])
+        @journey = @traveller.journeys.find(params[:id])
+
+    end
+
+    def journey_params
+        params.require(:journey).permit(:from, :to, :departure_date, :arrival_date, :capacity, :rate)
+    end
+
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:84]: remove extra spaces

File: app/controllers/orders_controller.rb [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+        # byebug
+        @sender = Sender.find(params[:sender_id])
+        @order = @sender.orders.build
+        session[:traveller_id] = params[:traveller_id].to_i
+
+    end
+
+    def create
+        # byebug
+        @sender = Sender.find(params[:sender_id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:18]: use before_action for this line

File: app/controllers/orders_controller.rb [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            redirect_to sender_orders_path, notice: "successfully created an order"
+        else
+            render "new"
+        end
+
+    end
+
+    def edit
+        # byebug
+        @sender = Sender.find(params[:sender_id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:34]: use before action

File: app/controllers/orders_controller.rb [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+        @order = @sender.orders.find(params[:id])
+    end
+
+    def update
+        @sender = Sender.find(params[:sender_id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:39]: use before action

File: app/controllers/orders_controller.rb [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+        @order = @sender.orders.find(params[:id])
+        if @order.update(order_params)
+            redirect_to sender_order_path(@order), notice: "successfully updated!"
+        else
+            render "edit"
+        end
+    end
+
+    def index
+        @sender = Sender.find(params[:sender_id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:49]: do we really need this? if yes use before action

File: app/controllers/orders_controller.rb [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+        @orders = @sender.orders
+    end
+
+    def show
+        @order = Order.find(param[:id])
+    end
+
+    def destroy
+        @order = Order.find(params[:id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:58]: use before action

File: app/controllers/orders_controller.rb [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+        end
+    end
+
+    def index
+        @sender = Sender.find(params[:sender_id])
+        @orders = @sender.orders
+    end
+
+    def show
+        @order = Order.find(param[:id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:54]: set it in before action

File: app/controllers/orders_controller.rb [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+        end
+    end
+
+    def cannot_edit_traveller_id_from_url
+        if session[:traveller_id] != params[:traveller_id].to_i
+            flash["alert"] = "you cannot edit the traveller id from url."
+            redirect_to traveller_list_path
+        end
+    end
+
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb:93]: remove extra spaces

File: app/controllers/senders_controller.rb [hunk 18]
```diff
@@ -0,0 +1,112 @@
+class SendersController < ApplicationController
+    before_action :already_sender, only: [:new]
+    before_action :sender_cannot_access_the_other_sender_info, only: [:edit, :show, :update, :destroy]
+    before_action :restrict_index_action, only: [:index]
+    before_action :restrict_traveller_to_access_sender
+
+    # include TravellerHelper
```
Comment (momnanaseer) [inline app/controllers/senders_controller.rb:7]: remove extra spaces

File: app/controllers/senders_controller.rb [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+        else
+            render "new"
+
+        end
+
+
+    end
+
+    def show
+        @sender = Sender.find(current_user_credential.user_id)
```
Comment (momnanaseer) [inline app/controllers/senders_controller.rb:30]: use find_by instead of find

File: app/controllers/senders_controller.rb [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+        # byebug
+    end
+
+    def edit
+        @sender = Sender.find(params[:id])
+    end
+
+    def update
+        @sender = Sender.find(params[:id])
```
Comment (momnanaseer) [inline app/controllers/senders_controller.rb:39]: use before actions

File: app/controllers/senders_controller.rb [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+            redirect_to sender_path(@sender)
+        else
+            render "edit"
+
+        end
+
+    end
+
+    def destroy
+        byebug
```
Comment (momnanaseer) [inline app/controllers/senders_controller.rb:50]: remove this byebug

File: app/controllers/senders_controller.rb [hunk 22]
```diff
@@ -0,0 +1,112 @@
...
+        @sender = Sender.find(params[:id])
+        @sender.destroy
+        current_user_credential.destroy
+        redirect_to root_path
+        
+    end
+
+    def index
+        byebug
```
Comment (momnanaseer) [inline app/controllers/senders_controller.rb:59]: remove this byebug

File: app/controllers/senders_controller.rb [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+    end
+
+    def sender_cannot_access_the_other_sender_info
+        if  current_user_credential.user_id != params[:id].to_i
+            flash["notice"] = "you'r not authorised to access it."
+            redirect_to root_path
+        end
+    end
+
+    def restrict_traveller_to_access_sender
```
Comment (momnanaseer) [inline app/controllers/senders_controller.rb:96]: this method is duplicate. move it to application controller and update your logic

File: app/controllers/travellers_controller.rb [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+    end
+
+    def orders
+        # byebug
+        @traveller = Traveller.find(current_user_credential.user_id)
+        @orders = @traveller.orders
+    end
+
+    def destroy
+        byebug
```
Comment (momnanaseer) [inline app/controllers/travellers_controller.rb:61]: remove byebug

File: app/javascript/controllers/from_controller.js [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+              
+              case "locality":
+                this.cityTarget.value = component.long_name;
+                break;
+
+              
+            }
+          }
+
+
```
Comment (momnanaseer) [inline app/javascript/controllers/from_controller.js:46]: remove extra spaces

File: app/models/journey.rb [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+  enum status: { not_completed: 0, completed: 1 }, _default: "not_completed"
+
+  validates :from, :to, :departure_date, :arrival_date, :capacity, :rate, presence: true
+  validates :capacity, numericality: { greater_than_or_equal_to: 5, less_than_or_equal_to: 40,  only_integer: true }
+
+
+  private
+
+  def self.search(from,to,capacity)
+    Journey.where(from: from.capitalize, to: to.capitalize).where("capacity > ?",capacity.to_i)
```
Comment (momnanaseer) [inline app/models/journey.rb:13]: use downcase instead of capitalize

File: app/views/devise/registrations/new.html.erb [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+  <div class="container">
+    <h1><%= t('.sign_up') %></h1>
+
+      <div class="row">
+        <div class="col-9">
+
+          <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  
+            <%
+=begin%>
```
Comment (momnanaseer) [inline app/views/devise/registrations/new.html.erb:11]: indent this file properly, remove extra spaces.

File: app/views/home/index.html.erb [hunk 28]
```diff
@@ -0,0 +1,27 @@
+<div class="container">
+
+    <h1>Welcome to Delivery App</h1>
+
+    <% if current_user_credential %>
+
+        <% if current_user_credential.user_type == nil %>
```
Comment (momnanaseer) [inline app/views/home/index.html.erb:7]: just use current_user.present? condition for this.

File: app/views/orders/edit.html.erb [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+        </div>
+        </div>
+        <% f.buttons do %>
+        <%= f.submit 'Save' %>
+        <% end %>
+    <% end %>
+    <% end %> 
+<%
+=end%>
+
```
Comment (momnanaseer) [inline app/views/orders/edit.html.erb:68]: do we really need this code?

File: app/views/travellers/orders.html.erb [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+                    <th scope="row">1</th>
+                    <td><%=current_user_credential.user.firstname%></td>
+                    <td><%=order.receiver_name%></td>
+                    <td>
+                        <%
+=begin%>
+ <%= link_to "view", traveller_journey_path(@traveller,journey), class: "btn btn-outline-info btn-sm"%>
+                        <%= link_to "edit", edit_traveller_journey_path(@traveller,journey), class: "btn btn-outline-success btn-sm" %>
+                        <%= link_to "delete", traveller_journey_path(@traveller,journey), method: :delete, class: "btn btn-outline-danger btn-sm" %> 
+<%
```
Comment (momnanaseer) [inline app/views/travellers/orders.html.erb:24]: do we really need this code?

File: app/views/travellers/show.html.erb [hunk 31]
```diff
@@ -0,0 +1,180 @@
+
+<div class="container">
+    <%
+=begin%>
+ <h1>welcome <%=@traveller.firstname%></h1>
+    <%= link_to "Edit", edit_traveller_path, class: "btn btn-outline-success my-2 my-sm-0 left-margin" %>
+    <%= link_to "Delete Profile", traveller_path(@traveller), method: :delete, class: "btn btn-outline-danger my-2 my-sm-0 left-margin" %> 
+<%
+=end%>
```
Comment (momnanaseer) [inline app/views/travellers/show.html.erb:9]: do we really need this code?

File: lib/templates/erb/scaffold/_form.html.erb [hunk 32]
```diff
@@ -0,0 +1,11 @@
+<%%= semantic_form_for @<%= singular_name %> do |f| %>
```
Comment (momnanaseer) [inline lib/templates/erb/scaffold/_form.html.erb:1]: what is this file used for?

File: app/controllers/home_controller.rb (hunk 2 above)
Comment (muhammadanas123) [reply app/controllers/home_controller.rb:2]: done

File: app/controllers/journeys_controller.rb (hunk 5 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:9]: done

File: app/controllers/journeys_controller.rb (hunk 6 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:23]: done

File: app/controllers/journeys_controller.rb (hunk 4 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:31]: done

File: app/controllers/journeys_controller.rb [hunk 33]
```diff
@@ -0,0 +1,101 @@
...
+    def destroy
+        @journey.destroy
+        redirect_to traveller_journeys_path
+
+    end
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:72]: done

File: app/controllers/journeys_controller.rb (hunk 7 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:76]: done

File: app/controllers/journeys_controller.rb (hunk 10 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:84]: done

File: app/controllers/orders_controller.rb (hunk 11 above)
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:18]: done

File: app/controllers/orders_controller.rb (hunk 12 above)
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:34]: done

File: app/controllers/orders_controller.rb (hunk 13 above)
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:39]: done

File: app/controllers/orders_controller.rb (hunk 14 above)
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:49]: done

File: app/controllers/orders_controller.rb (hunk 15 above)
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:58]: done

File: app/controllers/orders_controller.rb [hunk 34]
```diff
@@ -0,0 +1,97 @@
...
+        @order = Order.find(param[:id])
+    end
+
+    def destroy
+        @order = Order.find(params[:id])
+        @order.destroy
+        redirect_to sender_orders_path
+
+    end
+
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:63]: done

File: app/controllers/orders_controller.rb (hunk 17 above)
Comment (muhammadanas123) [reply app/controllers/orders_controller.rb:93]: done

File: app/controllers/senders_controller.rb (hunk 18 above)
Comment (muhammadanas123) [reply app/controllers/senders_controller.rb:7]: done

File: app/controllers/senders_controller.rb (hunk 19 above)
Comment (muhammadanas123) [reply app/controllers/senders_controller.rb:30]: done

File: app/controllers/senders_controller.rb (hunk 20 above)
Comment (muhammadanas123) [reply app/controllers/senders_controller.rb:39]: done

File: app/controllers/senders_controller.rb (hunk 21 above)
Comment (muhammadanas123) [reply app/controllers/senders_controller.rb:50]: done

File: app/controllers/senders_controller.rb (hunk 22 above)
Comment (muhammadanas123) [reply app/controllers/senders_controller.rb:59]: done

File: app/controllers/travellers_controller.rb (hunk 24 above)
Comment (muhammadanas123) [reply app/controllers/travellers_controller.rb:61]: done

File: app/javascript/controllers/from_controller.js (hunk 25 above)
Comment (muhammadanas123) [reply app/javascript/controllers/from_controller.js:46]: done

File: app/models/journey.rb (hunk 26 above)
Comment (muhammadanas123) [reply app/models/journey.rb:13]: done

File: app/views/devise/registrations/new.html.erb (hunk 27 above)
Comment (muhammadanas123) [reply app/views/devise/registrations/new.html.erb:11]: done

File: app/views/home/index.html.erb (hunk 28 above)
Comment (muhammadanas123) [reply app/views/home/index.html.erb:7]: this condition is for when a user is registered but not become a traveller or sender.

File: app/views/travellers/orders.html.erb (hunk 30 above)
Comment (muhammadanas123) [reply app/views/travellers/orders.html.erb:24]: done

File: app/views/travellers/show.html.erb (hunk 31 above)
Comment (muhammadanas123) [reply app/views/travellers/show.html.erb:9]: done

File: lib/templates/erb/scaffold/_form.html.erb (hunk 32 above)
Comment (muhammadanas123) [reply lib/templates/erb/scaffold/_form.html.erb:1]: no we dont need this file..removed

File: app/controllers/journeys_controller.rb (hunk 9 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb:55]: can i discuss this with you?

File: app/controllers/senders_controller.rb (hunk 23 above)
Comment (muhammadanas123) [reply app/controllers/senders_controller.rb:96]: this is sender's controller specific restriction.

File: app/controllers/journeys_controller.rb [hunk 35]
```diff
@@ -0,0 +1,79 @@
...
+    before_action :find_and_set_journey, only: [:show, :edit, :update, :destroy]
+    before_action :restrict_sender_to_access_journey
+    before_action :set_traveller, only: [:new, :create, :index]
+
+    def new
+        @journey = @traveller.journeys.build
+    end
+
+    def create 
+        byebug
```
Comment (momnanaseer) [inline app/controllers/journeys_controller.rb:12]: this byebug is still here? @muhammadanas123

File: app/views/journeys/new.html.erb [hunk 36]
```diff
@@ -0,0 +1,56 @@
+<div class="container">
```
Comment (momnanaseer) [inline app/views/journeys/new.html.erb:1]: make a _form partial and render it here, @muhammadanas123

File: app/views/orders/edit.html.erb [hunk 37]
```diff
@@ -0,0 +1,40 @@
+<div class="container">
```
Comment (momnanaseer) [inline app/views/orders/edit.html.erb:1]: make a _form partial and render it here. @muhammadanas123

File: app/views/senders/_form.html.erb [hunk 38]
```diff
@@ -0,0 +1,54 @@
...
+
+            <div class="form-group col-md-4">
+                <%= f.label :country %><br />
+                <%= f.text_field :country, autofocus: true, autocomplete: "state", class: "form-control" %>
+            </div>
+        </div>
+
+
+
+        <div class="actions">
```
Comment (momnanaseer) [inline app/views/senders/_form.html.erb:48]: extra spaces

File: app/views/senders/show.html.erb [hunk 39]
```diff
@@ -0,0 +1,170 @@
+<div class="container">
```
Comment (momnanaseer) [inline app/views/senders/show.html.erb:1]: this file contains too much lines of code, separate them out into partials and render them

File: app/views/travellers/_form.html.erb [hunk 40]
```diff
@@ -0,0 +1,54 @@
...
+                <%= f.number_field :phone_no, autocomplete: 'phone_no', class: 'form-control' %>
+            </div>
+
+            <div class="form-group col-md-6">
+                <%= f.label :landline %>
+                <%= f.number_field :landline, autocomplete: 'landline', class: 'form-control' %>
+            </div>
+
+        </div>
+
```
Comment (momnanaseer) [inline app/views/travellers/_form.html.erb:28]: extra space

File: app/views/travellers/_form.html.erb [hunk 41]
```diff
@@ -0,0 +1,54 @@
...
+            </div>
+
+            <div class="form-group col-md-4">
+                <%= f.label :country %><br />
+                <%= f.text_field :country, autofocus: true, autocomplete: "state", class: "form-control" %>
+            </div>
+        </div>
+
+
+
```
Comment (momnanaseer) [inline app/views/travellers/_form.html.erb:47]: extra space

File: app/views/travellers/show.html.erb [hunk 42]
```diff
@@ -0,0 +1,168 @@
+<div class="container">
+
```
Comment (momnanaseer) [inline app/views/travellers/show.html.erb:2]: this file contains too much lines of code, separate them out into partials.

File: app/controllers/journeys_controller.rb (hunk 35 above)
Comment (muhammadanas123) [reply app/controllers/journeys_controller.rb]: removed

File: app/views/senders/_form.html.erb (hunk 38 above)
Comment (muhammadanas123) [reply app/views/senders/_form.html.erb:48]: done

File: app/views/journeys/new.html.erb (hunk 36 above)
Comment (muhammadanas123) [reply app/views/journeys/new.html.erb:1]: sure

File: app/views/orders/edit.html.erb (hunk 37 above)
Comment (muhammadanas123) [reply app/views/orders/edit.html.erb:1]: sure

File: app/views/journeys/new.html.erb (hunk 36 above)
Comment (muhammadanas123) [reply app/views/journeys/new.html.erb:1]: done

File: app/views/orders/edit.html.erb (hunk 37 above)
Comment (muhammadanas123) [reply app/views/orders/edit.html.erb:1]: done

File: app/views/senders/show.html.erb (hunk 39 above)
Comment (muhammadanas123) [reply app/views/senders/show.html.erb:1]: done

File: app/views/travellers/_form.html.erb [hunk 43]
```diff
@@ -0,0 +1,54 @@
...
+            <%= f.text_field :firstname, autofocus: true, autocomplete: 'firstname', class: 'form-control' %>
+        </div>
+
+        <div class="form-group">
+            <%= f.label :lastname %>
+            <%= f.text_field :lastname, autocomplete: 'lastname', class: 'form-control' %>
+        </div>
+
+        
+
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply app/views/travellers/_form.html.erb]: done

File: app/views/travellers/_form.html.erb (hunk 40 above)
Comment (muhammadanas123) [reply app/views/travellers/_form.html.erb]: done

File: app/views/travellers/_form.html.erb (hunk 41 above)
Comment (muhammadanas123) [reply app/views/travellers/_form.html.erb]: done

File: app/views/travellers/_form.html.erb [hunk 44]
```diff
@@ -0,0 +1,54 @@
...
+            </div>
+        </div>
+
+
+
+        <div class="actions">
+            <%= f.submit(@traveller.new_record? ? "Register" : "Update" , class: "btn btn-outline-success" )%>
+        </div>
+        <% end %>
+
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply app/views/travellers/_form.html.erb]: done

File: app/views/travellers/show.html.erb (hunk 42 above)
Comment (muhammadanas123) [reply app/views/travellers/show.html.erb:2]: done

File: db/migrate/20221226082349_add_polymorphic_to_usercredentials.rb [hunk 45]
```diff
@@ -0,0 +1,10 @@
+class AddPolymorphicToUsercredentials < ActiveRecord::Migration[6.1]
+  def change
+
+    change_table :user_credentials do |t|
+      t.references :user, polymorphic: true
+
+    end
+
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply db/migrate/20221226082349_add_polymorphic_to_usercredentials.rb]: done

File: db/migrate/20221226082349_add_polymorphic_to_usercredentials.rb [hunk 46]
```diff
@@ -0,0 +1,10 @@
+class AddPolymorphicToUsercredentials < ActiveRecord::Migration[6.1]
+  def change
+
+    change_table :user_credentials do |t|
+      t.references :user, polymorphic: true
+
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply db/migrate/20221226082349_add_polymorphic_to_usercredentials.rb]: done

File: config/credentials.yml.enc [hunk 47]
```diff
@@ -1 +1 @@
-vpeJxSa6Xc1ElozoFTU54QIpBD84ntvPmcUC/ga8WQNLhevAHzmhgmB59QAGyAaGMS3ILo4d3McNrD1nptil9MzJumKz6SaSc1/Na95etY+TvCjJ30mslz6OCmQW0Pim2IBTIS5GfYvxM+JNsthpm6afrlc6voX+oSiKAO7rk+CdbBKohwvdORZ6Bm3WCabb4LVrzuBu6v3NAsVzG4LqtrqXaDVmIMeOPtLmTX187n1eYGe/zQQXatoUnBSKQ3gkou6AOU5TjDBYYU3eLHL084C9TzemCI8r2mfHEXp4mDb/IfLWWkgg7fVRjvYdEd1mNgJJbiXjPHPOz61FjIF1fHeibHPSne7A8F7+Uj1sYhZAps+nNLb2OACLGMJpnpVkUzyAFUQ5elcnljI/iM3yygSdsEPWVh+JUTWZ--NTQBMM1+Jx6mjip1--Rm+NY7NDgX4MI2klNsnNfQ==
\ No newline at end of file
+44YVnzsI9ye23CjOco7O5jwtKs+CVSBOM+WgnN6Rql7wXQDaWXAmtFPGswumSpqXBnVkgLBKrXbl37kknj+0aYB58lkUVbHzgf7xQjlGCnnMIs2Edq8I7CGTViYyNtkmlBQSZsoGjcRMXeeropCwIy38ufLzp8dqBiw7SvW3AD0RluCdVFiD7S2rDboHy5gnuGHaGC35a4cruJWYlaDrtYn8qAYZwk9mUgfW36xnFogmCCvaVj5VEqCQ12JtN7S9ZK3aVp/UBfTtVxViC3/sXZnO7qOrIOPSLX/x+nL7rLR+q8j66AaJBe1mPGBDIWghcXyx7sROS4Dy1aXx5dT2MCJ70fMGYGkqxWtaREEr8TIxGd+CyYJjdF8KQuW5FtJ9DoM9fDOTTZpXa7doBec6dMNfiMF0dxESe0Bf5YGm5DrP7aiMkvOoDTb064rlsBhDS55gVrSCtIW3BKhMBTsLQSevjLsXW49AU0BPItk=--5LnUknO+Om/z1KvA--W3AlspwHIInLjMzCzt97vQ==
```
[the remark this answers is not in the document]
Comment (muhammadanas123) [reply config/credentials.yml.enc:1]: sure

**Pull request merged**